pallets/liquidity-pools-gateway/src/lib.rs

```diff
@@ -188,10 +272,77 @@ pub mod pallet {
 		/// Emitted when you can `end_batch_message()` but the packing process
 		/// was not started by `start_batch_message()`.
 		MessagePackingNotStarted,
+
+		/// Invalid multi router.
+		InvalidMultiRouter,
+
+		/// Inbound domain session not found.
+		InboundDomainSessionNotFound,
+
+		/// The router that sent the inbound message is unknown.
+		UnknownInboundMessageRouter,
+
+		/// The router that sent the message is not the first one.
+		MessageExpectedFromFirstRouter,
+
+		/// The router that sent the proof should not be the first one.
+		ProofNotExpectedFromFirstRouter,
+
+		/// A message was expected instead of a proof.
+		ExpectedMessageType,
+
+		/// A message proof was expected instead of a message.
+		ExpectedMessageProofType,
+
+		/// Pending inbound entry not found.
+		PendingInboundEntryNotFound,
+
+		/// Message proof cannot be retrieved.
+		MessageProofRetrieval,
+
+		/// Recovery message not found.
+		RecoveryMessageNotFound,
 	}
 
 	#[pallet::call]
 	impl<T: Config> Pallet<T> {
+		/// Sets the router IDs used for a specific domain,
+		#[pallet::weight(T::WeightInfo::set_domain_routers())]
+		#[pallet::call_index(0)]
+		pub fn set_domain_routers(
+			origin: OriginFor<T>,
+			domain: Domain,
+			router_ids: BoundedVec<T::RouterId, T::MaxRouterCount>,
+		) -> DispatchResult {
+			T::AdminOrigin::ensure_origin(origin)?;
+
+			ensure!(domain != Domain::Centrifuge, Error::<T>::DomainNotSupported);
+
+			//TODO(cdamian): Outbound -  Call router.init() for each router?
```

Comment:
@lemunozm Just to be ultra-sure. There's no `init` logic in the refactored routers right?

Reply:
Exists when configures a router in `pallet-axelar-router` under `set_config()`

Reply:
OK, but the gateway won't have anything to do with that. Thank you!